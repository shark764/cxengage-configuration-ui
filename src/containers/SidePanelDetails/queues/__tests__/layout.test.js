/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import QueuesDetailsPanel from '../layout';

describe('<queuesDetailsPanel />', () => {
  let customItem;
  beforeEach(() => {
    customItem = {
      name: 'Test',
      description: 'Test'
    };
  });
  it('renders queues detailsPanel', () => {
    const rendered = shallow(
      <QueuesDetailsPanel
        id="0000-0000-0000-0000-0000"
        className="details-panel"
        userHasUpdatePermission={true}
        children={'Mock Child'}
        item={customItem}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders queues detailsPanel with no update permision', () => {
    const rendered = shallow(
      <QueuesDetailsPanel
        id="0000-0000-0000-0000-0000"
        className="details-panel"
        userHasUpdatePermission={false}
        children={'Mock Child'}
        item={customItem}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
