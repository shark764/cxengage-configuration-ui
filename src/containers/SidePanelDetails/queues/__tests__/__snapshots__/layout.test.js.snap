// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<queuesDetailsPanel /> renders queues detailsPanel 1`] = `
<styled.div
  className="dtpanel-entity-queues"
>
  Mock Child
</styled.div>
`;

exports[`<queuesDetailsPanel /> renders queues detailsPanel with no update permision 1`] = `
<styled.div
  className="dtpanel-entity-queues"
>
  Mock Child
</styled.div>
`;
